server/metric/mock_metric_test.go

```diff
@@ -209,3 +214,155 @@ func TestMockMetricServer_ListMetricDescriptors(t *testing.T) {
 		t.Errorf("ListMetricDescriptors(%q) == %q, want %q", in, response, want)
 	}
 }
+
+func TestMockMetricServer_GetMetricDescriptorError(t *testing.T) {
+	in := &monitoring.GetMetricDescriptorRequest{}
+	want := validation.ErrMissingField.Err()
+	missingFields := map[string]struct{}{"Name": {}}
+	response, err := client.GetMetricDescriptor(ctx, in)
+	if err == nil {
+		t.Errorf("GetMetricDescriptor(%q) == %q, expected error %q", in, response, want)
+	}
+
+	if !strings.Contains(err.Error(), want.Error()) {
```

Comment:
I had to look into the code to figure out why we weren't doing:
``` if err != validation.ErrMissingField { ....```

There's a couple issues here:
1) `ErrMissingField` shouldn't start with `Err` if it isn't a constant `error`
I know it's not related to this PR, `validation.ErrMissingField` has the naming convention of an error constant like `io.ErrUnexpectedEOF`. I had to look into the code to realize that it is actually a grpc status object.

2) Using string comparison to compare errors is brittle.
String comparison in tests is fine. But it's very dangerous for production code -- refactoring's can accidentally break things without a compile-time error. If we don't need check if an error is of type `MissingField` outside of tests, then we shouldn't export `ErrMissingField`. If its a useful thing to do, then check out the patterns here: https://blog.golang.org/go1.13-errors.

Reply:
As this doesn't pertain too much to this PR specifically, I've created an [issue](https://github.com/googleinterns/cloud-operations-api-mock/pull/10#discussion_r443842037) to be addressed  another PR.